bigtable/google/cloud/bigtable/table.py

```diff
@@ -618,6 +634,83 @@ def _do_mutate_retryable_rows(self):
         return self.responses_statuses
 
 
+class _ClusterState(object):
```

Comment:
Why is this class private?  We use instances of it as the values of the mapping returned by `Table.get_cluster_states` (a public API wrapper), which should logically make the class itself public.

Reply:
1)Agree with you _ClusterState should be public. Will change it to ClusterState. This class does not mock ptobuf. 
2)I should not remove 
def _ClusterStatePB(replication_state):
    from google.cloud.bigtable_admin_v2.proto import (
        table_pb2 as table_v2_pb2)

    return table_v2_pb2.Table.ClusterState(
        replication_state=replication_state
    )

which is used for costructing protobuf

response_pb = _TablePB(
            cluster_states={'cluster-id1': _ClusterStatePB(INITIALIZING),
                            'cluster-id2': _ClusterStatePB(
                                PLANNED_MAINTENANCE),
                            'cluster-id3': _ClusterStatePB(READY),
                            },
        )

Do you agree? removing def _ClusterStatePB will not make any significant change because we need 

table_v2_pb2.Table.ClusterState(
        replication_state=replication_state
    )

to construct _TablePB

Reply:
@tseaver The only thought here is that user is getting exposed to object of class `_ClusterState` via `Table.get_cluster_states` as an `OutputOnly` value. User can't set `replication_state` manually for any rpc, so user should have no need to create an a `_ClusterState` object manually.

Reply:
@AVaksman We still need to document the things they are getting handed.

Reply:
@vikas-jamdar Please do rename `_ClusterState` to `ClusterState`.  You can leave the factory function in the test module as-is.

Reply:
@tseaver understood, thank you. I was just explaining why it was proposed as a private class.